internal/services/eventhub/eventhub_resource.go

```diff
@@ -51,11 +52,11 @@ func resourceEventHub() *pluginsdk.Resource {
 				ValidateFunc: validate.ValidateEventHubName(),
 			},
 
-			"namespace_name": {
+			"namespace_id": {
 				Type:         pluginsdk.TypeString,
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: validate.ValidateEventHubNamespaceName(),
+				ValidateFunc: namespaces.ValidateNamespaceID,
 			},
 
 			"resource_group_name": commonschema.ResourceGroupName(),
```

Comment:
@catriona-m as pointed out by @CorrenSoft this probably wants to be deprecated as well